File: software/examples/perl/example_interrupt.pl
#!/usr/bin/perl

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::BrickletIO4;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your IO-4 Bricklet

# Callback subroutine for interrupt callback
sub cb_interrupt
{
    my ($interrupt_mask, $value_mask) = @_;

    print "Interrupt Mask: " . sprintf('%04b', $interrupt_mask) . "\n";
    print "Value Mask: " . sprintf('%04b', $value_mask) . "\n";
    print "\n";
}

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $io = Tinkerforge::BrickletIO4->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Register interrupt callback to subroutine cb_interrupt
$io->register_callback($io->CALLBACK_INTERRUPT, 'cb_interrupt');

# Enable interrupt on pin 0
$io->set_interrupt(1 << 0);

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();

File: software/examples/javascript/ExampleInterrupt.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your IO-4 Bricklet

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var io = new Tinkerforge.BrickletIO4(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Enable interrupt on pin 0
        io.setInterrupt(1 << 0);
    }
);

// Register interrupt callback
io.on(Tinkerforge.BrickletIO4.CALLBACK_INTERRUPT,
    // Callback function for interrupt callback
    function (interruptMask, valueMask) {
        console.log('Interrupt Mask: ' + interruptMask.toString(2));
        console.log('Value Mask: ' + valueMask.toString(2));
        console.log();
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
